src/test/cluster_health.zig

```diff
@@ -0,0 +1,117 @@
+const std = @import("std");
+const assert = std.debug.assert;
+const mem = std.mem;
+const log = std.log.scoped(.health);
+
+pub const ClusterHealthOptions = struct {
+    replica_count: u8,
+    /// The minimum number of replicas that must be up.
+    //replica_count_min: u8,
+    seed: u64,
+
+    /// Probability per tick that a crash will occur.
+    crash_probability: f64,
+    /// Minimum duration of a crash.
+    crash_stability: u32,
+    /// Probability per tick that a crashed replica will recovery.
+    restart_probability: f64,
+    /// Minimum time a replica is up until it is crashed again.
+    restart_stability: u32,
+};
+
+const ReplicaHealth = union(enum) {
+    /// When >0, the replica cannot crash.
+    /// When =0, the replica may crash.
+    up: u32,
+    /// When >0, this is the ticks remaining until recovery is possible.
+    /// When =0, the replica may recover.
+    down: u32,
+};
+
+pub const ClusterHealth = struct {
+    options: ClusterHealthOptions,
+    prng: std.rand.DefaultPrng,
+
+    /// Health by replica.
+    health: []ReplicaHealth,
+
+    pub fn init(allocator: mem.Allocator, options: ClusterHealthOptions) !ClusterHealth {
+        assert(options.replica_count > 0);
+        assert(options.crash_probability < 1.0);
+        assert(options.crash_probability >= 0.0);
+        assert(options.restart_probability < 1.0);
+        assert(options.restart_probability >= 0.0);
+
+        const health = try allocator.alloc(ReplicaHealth, options.replica_count);
+        errdefer allocator.dealloc(health);
+        mem.set(ReplicaHealth, health, .{.up = 0});
```

Comment:
We could run `zig fmt src` to format everything. It will add a space before and after the `.up = 0`.